packages/eas-cli/src/submit/ios/AppProduce.ts

```diff
@@ -81,9 +80,11 @@ async function createAppStoreConnectAppAsync(
       bundleIdentifier: bundleId,
     });
   } else {
-    Log.warn(
-      `Provisioning is not available for user "${authCtx.appleId}", skipping bundle identifier check.`
-    );
+    const entity = isUserAuthCtx(authCtx)
```

Comment:
If I understand correctly asc api key does not support everything for submit, I might be wrong, but currently only creating new app (or maybe querying) on appstore connect requires a user login. We should probably handle that case before it errors with some cryptic error.

Another issue is that it might be weird if users do not have asc key configured and they have ascAppId in eas.json and will ask to provide asc API key even if they already authenticated with asc API key to login to apple

Both above points do not introduce any regressions on current behavior, so it's probably fine to handle them in separate PR